### PR TITLE
Clear backstack when logging in/out

### PR DESCRIPTION
**Pull request by willsb**
Before: You can hit the back button and see the login page/onboarding after logging in. You can see the settings/log after logging out.

Now: The backstack behaves appropriately

Squash whenever.

### CODE REVIEW

File: Toggl.Giskard/Presenters/TogglPresenter.cs [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+            var intent = base.CreateIntentForRequest(request);
+
+            if (clearBackStackTypes.Contains(request.ViewModelType))
+            {
+                intent.SetFlags(ActivityFlags.ClearTop | ActivityFlags.ClearTask | ActivityFlags.NewTask);
+            }
+
+            return intent;
+        }
+
```
Comment (kornelijepetak) [inline Toggl.Giskard/Presenters/TogglPresenter.cs]: One `\n` too much

File: Toggl.Giskard/Presenters/TogglPresenter.cs [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
 using MvvmCross.Droid.Support.V7.AppCompat;
+using Toggl.Foundation.MvvmCross.ViewModels;
 using Toggl.Foundation.MvvmCross.ViewModels.Hints;
 using Toggl.Giskard.Activities;
 
 namespace Toggl.Giskard.Presenters
 {
     public sealed class TogglPresenter : MvxAppCompatViewPresenter
     {
+        private readonly HashSet<Type> clearBackStackTypes = new HashSet<Type>
```
Comment (kornelijepetak) [inline Toggl.Giskard/Presenters/TogglPresenter.cs]: This is a number of types small enough that the array would suffice, and arrays are IMO a bit easier on the eyes.
No need to change, just saying.

Comment (willsb) [reply Toggl.Giskard/Presenters/TogglPresenter.cs]: I like my O(1), so I'm keeping this 💃

Comment (kornelijepetak) [reply Toggl.Giskard/Presenters/TogglPresenter.cs]: Sure. Like I said, for small number it's irrelevant. Not only that. This happens so rarely that we already talked too much about it. 😄

**Pull request merged**